src/test/java/com/microsoft/sqlserver/jdbc/NTLMConnectionTest.java

```diff
@@ -0,0 +1,689 @@
+     * [filled in test]                                                             // computer name
+     * 5, 0,                                                                        // MsvAvDnsTreeName avid
+     * 18, 0,                                                                       // MsvAvDnsTreeName avlen
+     * 103, 0, 97, 0, 108, 0, 97, 0, 120, 0, 121, 0, 46, 0, 97, 0, 100, 0,          // tree name
+     * 7, 0,                                                                        // MsvAvTimestamp avid
+     * 8, 0,                                                                        // MsvAvTimestamp avlen
+     * 122, -115, 18, 50, -5, -32, -44, 1,                                          // timestamp
+     * 0, 0, 0, 0                                                                   // MsvAvEOL avid
+     * };
+     * 
+     * challengeTokenPart1 is everything up to targetinfo
+     * challengeTargetInfo1 is everything in targetinfo up to MsvAvDnsComputerName id
+     * challengeTargetInfo2 is everything in targetinfo after computer (server) name
+     * 
+     * NTLM_CHALLENGE_*_OFFSETS are defined offsets for the fields in the challenge token
+     * </pre>
+     */
+    private byte[] challengeToken1stPart = {78, 84, 76, 77, 83, 83, 80, 0, 2, 0, 0, 0, 12, 0, 12, 0, 56, 0, 0, 0, 21,
+            -126, -127, 2, 78, -76, 68, 118, 41, -30, -71, 100, 0, 0, 0, 0, 0, 0, 0, 0, -88, 0, -88, 0, 68, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 71, 0, 65, 0, 76, 0, 65, 0, 88, 0, 89, 0};
+
+    private byte[] challengeTargetInfo1 = {2, 0, 12, 0, 71, 0, 65, 0, 76, 0, 65, 0, 88, 0, 89, 0, 1, 0, 30, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 20, 0, 68, 0, 79, 0,
+            77, 0, 65, 0, 73, 0, 78, 0, 78, 0, 65, 0, 77, 0, 69, 0, 3, 0};
+
+    private byte[] challengeTargetInfo2 = {5, 0, 18, 0, 103, 0, 97, 0, 108, 0, 97, 0, 120, 0, 121, 0, 46, 0, 97, 0, 100,
+            0, 7, 0, 8, 0, 122, -115, 18, 50, -5, -32, -44, 1, 0, 0, 0, 0};
+
+    private final int NTLM_CHALLENGE_SIGNATURE_OFFSET = 0;
+    private final int NTLM_CHALLENGE_MESSAGETYPE_OFFSET = 8;
+
+    private final int NTLM_CHALLENGE_TARGETINFOLEN_OFFSET = 40;
+    private final int NTLM_CHALLENGE_TARGETINFO_OFFSET = 68;
+
+    // offsets for targetinfo av pairs
+    private final int NTLM_CHALLENGE_MSVAVDNSDOMAINNAME_OFFSET = NTLM_CHALLENGE_TARGETINFO_OFFSET + 54;
+    private final int NTLM_CHALLENGE_MSVAVDNSCOMPUTERNAME_OFFSET = NTLM_CHALLENGE_TARGETINFO_OFFSET + 78;
+    private final int NTLM_CHALLENGE_MSVAVTIMESTAMP_OFFSET = -16;
+
+    @Test
+    public void testNTLMBadSignature() throws SQLException {
+        if (ntlmPropsDefined) {
+            try (SQLServerConnection con = (SQLServerConnection) DriverManager.getConnection(connectionStringNTLM)) {
+                getServerFqdn(con);

```

Comment:
There are lots of duplicated code in `testNTLMBad*`, please move them into a common method.